File: perfect_sound_mobile/lib/models/Rating/RatingUpsertRequest.dart
class RatingUpsertRequest {
  int? songAndPodcastId;
  int? userId;
  double? ratingValue;

  RatingUpsertRequest({this.songAndPodcastId, this.userId, this.ratingValue});

  RatingUpsertRequest.fromJson(Map<String, dynamic> json) {
    songAndPodcastId = json['songAndPodcastId'];
    userId = json['userId'];
    ratingValue = json['ratingValue'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['songAndPodcastId'] = this.songAndPodcastId;
    data['userId'] = this.userId;
    data['ratingValue'] = this.ratingValue;
    return data;
  }
}

File: PerfectSound/PerfectSound.Model/Requests/Rating/RatingUpsertRequest.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace PerfectSound.Model.Requests.Rating
{
    public class RatingUpsertRequest
    {
        public int? SongAndPodcastId { get; set; }
        public int? UserId { get; set; }
        public double? RatingValue { get; set; }
    }
}
